gef.py

```diff
@@ -10597,10 +10597,34 @@ def read_cstring(self,
         encoding = encoding or "unicode-escape"
         length = min(address | (DEFAULT_PAGE_SIZE-1), max_length+1)
 
+        while not self.get_section(address + length - 1) and length > 0:
```

Comment:
what if the next section is also readable? I think we probably want to support that.